### PR TITLE
Transparency support

### PR DESCRIPTION
**Pull request by jswark**

### CODE REVIEW

File: shaders/lights.h [hunk 1]
```diff
@@ -17,11 +19,240 @@ struct Light
...
+    float4 points[4];
     float4 color;
     float4 normal;
     int type;
     float pad0;
     float pad2;
     float pad3;
 };
+
+struct SphQuad
```
Comment (ikryukov) [inline shaders/lights.h:31]: link to original paper

File: shaders/lights.h [hunk 2]
```diff
@@ -17,11 +19,240 @@ struct Light
...
+    float hv = h0 + v * (h1 - h0);
+    float hv2 = hv*hv;
+    float eps = 1e-5;
+    float yv = (hv < 1 - eps) ? (hv * d)/sqrt(1 - hv2) : squad.y1;
+
+    // 4. transform (xu, yv, z0) to world coords
+    return (squad.o + xu * squad.x + yv*squad.y  + squad.z0*squad.z);
+}
+
+float2 concentricSampleDisk(float2 u) {
```
Comment (ikryukov) [inline shaders/lights.h:135]: link to pbrt?

File: shaders/pathtracerMdl.hlsl [hunk 3]
```diff
@@ -295,6 +49,23 @@ Ray generateCameraRay(uint2 pixelIndex)
     return ray;
 }
 
+float3 offset_ray(const float3 p, const float3 n)
```
Comment (ikryukov) [inline shaders/pathtracerMdl.hlsl:52]: offsetRay

File: include/scene/scene.h [hunk 4]
```diff
@@ -123,7 +125,7 @@ class Scene
     // GPU side structure
     struct Light
```
Comment (ikryukov) [inline include/scene/scene.h:126]: GenericLight?

File: include/scene/scene.h [hunk 5]
```diff
@@ -123,7 +125,7 @@ class Scene
...
     {
-        glm::float4 points[16];
+        glm::float4 points[4];
```
Comment (ikryukov) [inline include/scene/scene.h:128]: data? it is not only points

**Review by ikryukov — CHANGES_REQUESTED** [review]

File: include/scene/scene.h (hunk 5 above)
Comment (jswark) [reply include/scene/scene.h:128]: I store data apart from points:

```
glm::float4 points[4];
glm::float4 color = glm::float4(1.0f);
glm::float4 normal;
int32_t type;
```

**Pull request merged**